languageserver/server/server.go

```diff
@@ -541,6 +540,10 @@ func (s *Server) DidChangeTextDocument(
 		Version: version,
 	}
 
+	// todo implement smarter cache invalidation with dependency resolution using https://github.com/onflow/cadence/pull/1634
+	// we should build dependency tree upfront and then based on the changes only reset checkers contained in that tree
+	s.checkers = make(map[common.LocationID]*sema.Checker)
```

Comment:
It's cheaper to clear the map than to allocate a new map each time:
```suggestion
	for locationID := range s.checkers {
		delete(s.checkers, locationID)
	}
```

Reply:
nice